_posts/2021-07-08-backup-restore-che-installation.md

```diff
@@ -0,0 +1,243 @@
+---
+title: Back up and Restore Eclipse Che Installation
+layout: post
+author: Mykola Morhun
+description: >-
+  How to create backups and do recovery of Eclipse Che
+categories: []
+keywords: ['backup', 'back up', 'restore']
+slug: /@mmorhun/backup-restore-che-installation
+---
+
+### Introduction
+
+Any application that runs in production should be backed up regularly.
+Even if the application runs inside Kubernetes or Openshift cluster.
+To back up an application in a Kubernetes or Openshift cluster, cluster admin should back up all the resources and definitions that the application owns and uses.
+It could be pretty easy in case when the application is, for example, a deployment with attached volume.
+But what if the application has a lot of objects to back up?
+In such case the task becomes much more complicated and will require from the admin understanding on how the components of the application work and interact with each other.
+Or... just back up the whole cluster, however, such approach has a lot of overhead.
+
+To address the backup / restore problem in [Eclipse Che](https://www.eclipse.org/che/), backup / restore feature was implemented.
+With it, cluster admin doesn't have to be aware of Eclipse Che internals in order to create a backup or do recovery of Che.
+Eclipse Che (Eclipse Che operator to be more precise) can create backups and restore the installation even if Che cluster was completely deleted!
+(Che operator should be available, though).
+
+Let me show you how easy the process of backing up and restoring Che is now.
+But first, let's talk about backup servers a bit.
+
+### Internal vs external backup server
+
+When all data for backup is gathered into a snapshot, then it is encrypted and sent to a backup server.
+The backup server should be set up beforehand and be accessible from within the cluster.
+This step requires choosing the backup server type and manual configuration of it.
+
+To make life a bit easier, Eclipse Che can automatically set up and configure a backup server in the same cluster.
+Such approach requires no additional configuration as everything is automated, but the main downside of it is that backups are stored in the same cluster and even the same namespace as Eclipse Che.
+
+Note, for production use, it is recommended to set up a backup server outside of the cluster.
```

Comment:
```suggestion
Note, for production use, it is recommended to set up a backup server in a distinct Kubernetes cluster.
```

Reply:
In general, it doesn't have to be in a cluster at all. We support AWS S3 or a user can host own backup server which is not part of a cluster...